frontend/src/devicelist/__tests__/DevicesTable.test.jsx

```diff
@@ -9,40 +9,238 @@ import {
+      const result = wrapper.instance().getComponents();
+      const props = wrapper.props();
+
+      /*  eslint-disable react/jsx-props-no-spreading */
+      const expectedRenderFunction = function Components(passedProps) {
+        return (
+          <div className="lightestGrey">
+            <MTableToolbar {...passedProps} />
+          </div>
+        );
+      };
+
+      const expectedRenderResult = expectedRenderFunction(props);
+      expect(result.Toolbar(props)).toMatchObject(expectedRenderResult);
+    });
+  });
+
+  describe("getColumnInfo() function", () => {
+    let result;
+
     beforeEach(() => {
-      wrapper = Enzyme.shallow(
-        <DevicesTable title={dummyTitle} devices={dummyDevices} />
+      result = wrapper.instance().getColumnInfo();
+    });
+
+    it("should return the expected column info to be used in the <MaterialTable/> component", () => {
+      const expected = [
+        {
+          title: "Name",
+          field: "name"
+        },
+        {
+          title: "Serial Number",
+          field: "serialNumber"
+        },
+        {
+          title: "Status",
+          field: "status",
+          lookup: {
+            Online: "Online",
+            Pending: "Pending",
+            Error: "Error",
+            Offline: "Offline"
+          }
+        },
+        {
+          title: "Private IP Address",
+          field: "privateIp"
+        },
+        {
+          title: "Public IP Address",
+          field: "publicIp"
+        },
+        {
+          title: "Actions",
+          field: "action",
+          filtering: false,
+          sorting: false,
+          align: "center",
+          export: false
+        }
+      ];
+      expect(result).toMatchObject(expected);
+    });
+    it(`should have a render() function that returns a <StatusIndicator/> component in Status column`, () => {
+      const dummyData = {
+        status: "Online"
+      };
+
+      const expectedRenderFunction = function Status(rowData) {
+        return <StatusIndicator status={rowData.status} />;
+      };
+
+      const expectedRenderResult = expectedRenderFunction(dummyData);
+      expect(result[2].render(dummyData)).toMatchObject(expectedRenderResult);
+    });
+    it(`should have a render() function that returns a <DeviceDetailsButton/> component in Actions column`, () => {
+      const sampleDevice = SampleData.getSampleSender();
+
+      const expectedRenderFunction = function Actions(rowData) {
+        return <DeviceDetailsButton deviceInfo={rowData} />;
+      };
+
+      const expectedRenderResult = expectedRenderFunction(sampleDevice);
+      expect(result[5].render(sampleDevice)).toMatchObject(
+        expectedRenderResult
       );
     });
-    it("contains one Box component", () => {
-      expect(wrapper.find(Box)).toHaveLength(1);
+  });
+
+  describe("getDetailPanel() function", () => {
+    let result;
```

Comment:
Rename this to `detailPanel` otherwise i don't really know what this variable is in the tests. Same for the `result` variables in other tests.